### ISSUE TITLE
Unreadable NiFTI predictions

### ISSUE THREAD

**Issue by mdbloice** [MIC-DKFZ/nnUNet#1322]
Hi there,

I am having issues with the NiFTI files being produced by nnUNet during inference. For example, if I open a NiFTI prediction, neither MITK Workbench nor ITK-SNAP recognise the format of the files produced by nnUNet, e.g:

![image](https://user-images.githubusercontent.com/16042756/223672602-5b00e2ae-865f-4d6c-be47-4a98b2af7789.png)

Has anyone any ideas about what could be happening? 

Any help would be appreciated! 

---

One note I would make: when I start training, I get the following warning about my training data set:

`WARNING: Not all images in the dataset have the same axis ordering. We very strongly recommend you correct that by reorienting the data. fslreorient2std should do the trick`

Could this be having an influence? I have not yet tried using `fslreorient2std` to correct my training data.

**Comment by FabianIsensee** [comment]
I think this is related to SimpleITK. What happens if you read a file with SimpleITK and save it again. Is it then also corrupted?
There are some files, for example from the ACDC challenge which are already corrupt the way they are shipped. Some ITK versions cannot read them, others can. I think that the ITK version in ITK snap might just be an unfortunate one...

**Comment by mdbloice** [comment]
Hi @FabianIsensee thanks for responding. 

In the end I tried a different machine, and the problems disappeared. So it must have been the particular setup of the first machine which caused the problems (everything else was the same: the exact same data, the exact same scripts were run, etc.)

I also found that even the masks created by `nnUNet_plan_and_preprocess` were corrupted (so before any training or inference), for example those stored in `./nnUNet_preprocessed/.../gt_segmentations/`, could not be read by ITK-SNAP or MITK.
